### PR TITLE
Add deploy step to CI

### PR DESCRIPTION
**Pull request by juliaogris**
Add deploy to GCP bucket step to CI so that PR changes can be live
reviewed.

- Use relative paths in index.html so that frontend can be served from non-root URL path (e.g. bucket)
- Add CI job `deploy` that rebuilds frontend without linting and copies result to storage bucket
- Add GitHub action to comment on PR with preview URL

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -6,11 +6,11 @@
 
     <title>Svelte app</title>
 
-    <link rel="icon" type="image/png" href="/favicon.png" />
-    <link rel="stylesheet" href="/global.css" />
-    <link rel="stylesheet" href="/build/bundle.css" />
+    <link rel="icon" type="image/png" href="favicon.png" />
+    <link rel="stylesheet" href="global.css" />
+    <link rel="stylesheet" href="bundle.css" />
```
Comment (camh-) [inline frontend/public/index.html:11]: Why did you move the files out of the build directory?

Comment (juliaogris) [reply frontend/public/index.html:11]: oh thanks for looking, couldn't even ask for a review ;)

I started this work as frontend cleanup/initial PR.
I did it because I think I kinda like things un-nested if there are that few. 
I know this mixes generated with manually created content, but I thought the tradeoff is ok, compared to the extra logic I had to add at other places to move the result files into one directory...
